### PR TITLE
Fixes ling notes copying on absorb

### PR DESCRIPTION
**Pull request by PKPenguin321**
Was working on this for my last ling fix PR and it got merged right before committing

:cl:
fix: Lings will copy the notes of people they absorb (including PDA codes, antag objectives, and objective conspirators) properly.
/:cl:

### CODE REVIEW

**Comment by PKPenguin321** [comment]
I'm just restoring old and broken functionality

**Comment by MrDoomBringer** [comment]
When did this get broken? I had a ling succ my PDA code like 2 days ago

**Comment by wesoda25** [comment]
How long was it broken for? Probably has do to with me barely getting ling anymore but I don’t remember this being a thing.

**Comment by MrDoomBringer** [comment]
@wesoda25 A couple days ago, I was a traitor and didn't spend any TC. Lexia the ling (surprise surprise) killed and DNA-Absorbed me, got my PDA code from my memories, and used it to buy sleeping carp.

This would make me think that the functionality PK is referring to isn't broken, unless I'm missing something.

**Comment by PKPenguin321** [comment]
Hm. Didn't seem to work when I tested it locally. Might have to test on a live server

**Comment by RogueSteampunker** [comment]
Can we... please just choose not to do this?

Broken or not, I'd like to believe that giving lings a method to find near everything out about a traitor, their pda, and otherwise is just asking for "ling is just overpowered tator" to get enforced twicefold once more.

File: code/modules/antagonists/changeling/powers/absorb.dm [hunk 1]
```diff
@@ -59,11 +59,31 @@
...
 
-	if(target.mind)//if the victim has got a mind
+	if(target.mind && user.mind)//if the victim and user have minds
 		// Absorb a lizard, speak Draconic.
 		user.copy_known_languages_from(target)
 
-		target.mind.show_memory(user, 0) //I can read your mind, kekeke. Output all their notes.
+		var/datum/mind/suckedbrain = target.mind
+		user.mind.memory += "<BR><b>We've absorbed [target]'s memories into our own...</b><BR>[suckedbrain.memory]<BR>"
+		if(LAZYLEN(suckedbrain.antag_datums))//this entire massive block is for antag memories and objectives
```
Comment (Cyberboss) [inline code/modules/antagonists/changeling/powers/absorb.dm]:
```suggestion
		if(LAZYLEN(suckedbrain.antag_datums))//this entire massive block is for antag memories and objectives
```

and unindent the stuff below

File: code/modules/antagonists/changeling/powers/absorb.dm [hunk 2]
```diff
@@ -59,11 +59,31 @@
...
+			for(var/datum/antagonist/antag_types in suckedbrain.antag_datums)
+				var/list/all_objectives = list()
+				all_objectives |= antag_types.objectives
```
Comment (Cyberboss) [inline code/modules/antagonists/changeling/powers/absorb.dm]:
```suggestion
```

File: code/modules/antagonists/changeling/powers/absorb.dm [hunk 3]
```diff
@@ -59,11 +59,31 @@
...
+	if(target.mind && user.mind)//if the victim and user have minds
 		// Absorb a lizard, speak Draconic.
 		user.copy_known_languages_from(target)
 
-		target.mind.show_memory(user, 0) //I can read your mind, kekeke. Output all their notes.
+		var/datum/mind/suckedbrain = target.mind
+		user.mind.memory += "<BR><b>We've absorbed [target]'s memories into our own...</b><BR>[suckedbrain.memory]<BR>"
+		if(LAZYLEN(suckedbrain.antag_datums))//this entire massive block is for antag memories and objectives
+			for(var/datum/antagonist/antag_types in suckedbrain.antag_datums)
+				var/list/all_objectives = list()
```
Comment (Cyberboss) [inline code/modules/antagonists/changeling/powers/absorb.dm]:
```suggestion
				var/list/all_objectives = antag_types.objectives.Copy()

File: code/modules/antagonists/changeling/powers/absorb.dm [hunk 4]
```diff
@@ -59,11 +59,31 @@
...
+				all_objectives |= antag_types.objectives
+				if(antag_types.antag_memory)
+					user.mind.memory += "[antag_types.antag_memory]<BR>"
+				if(LAZYLEN(all_objectives))
+					user.mind.memory += "<B>Objectives:</B>"
+					var/obj_count = 1
+					for(var/datum/objective/objective in all_objectives)
```
Comment (Cyberboss) [inline code/modules/antagonists/changeling/powers/absorb.dm]:
```suggestion
					for(var/O in all_objectives)
						var/datum/objective/objective = O
```

File: code/modules/antagonists/changeling/powers/absorb.dm [hunk 5]
```diff
@@ -59,11 +59,31 @@
...
+						user.mind.memory += "<br><B>Objective #[obj_count++]</B>: [objective.explanation_text]"
+						var/list/datum/mind/other_owners = objective.get_owners() - suckedbrain
+						if(other_owners.len)
+							user.mind.memory += "<ul>"
+							for(var/datum/mind/M in other_owners)
```
Comment (Cyberboss) [inline code/modules/antagonists/changeling/powers/absorb.dm]:
```suggestion
							for(var/mind in other_owners)
								var/datum/mind/M = mind
```

File: code/modules/antagonists/changeling/powers/absorb.dm [hunk 6]
```diff
@@ -59,11 +59,31 @@
...
-	if(target.mind)//if the victim has got a mind
+	if(target.mind && user.mind)//if the victim and user have minds
 		// Absorb a lizard, speak Draconic.
 		user.copy_known_languages_from(target)
 
-		target.mind.show_memory(user, 0) //I can read your mind, kekeke. Output all their notes.
+		var/datum/mind/suckedbrain = target.mind
+		user.mind.memory += "<BR><b>We've absorbed [target]'s memories into our own...</b><BR>[suckedbrain.memory]<BR>"
+		if(LAZYLEN(suckedbrain.antag_datums))//this entire massive block is for antag memories and objectives
+			for(var/datum/antagonist/antag_types in suckedbrain.antag_datums)
```
Comment (Cyberboss) [inline code/modules/antagonists/changeling/powers/absorb.dm]:
```suggestion
			for(var/A in suckedbrain.antag_datums)
				var/datum/antagonist/antag_types = A
```

File: code/modules/antagonists/changeling/powers/absorb.dm [hunk 7]
```diff
@@ -77,7 +97,6 @@
 				var/list/reversed = log_source[log_type]
 				if(islist(reversed))
 					say_log = reverseRange(reversed.Copy())
-					break
```
Comment (Cyberboss) [inline code/modules/antagonists/changeling/powers/absorb.dm]: Intentional?

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/changeling/powers/absorb.dm (hunk 7 above)
Comment (PKPenguin321) [reply code/modules/antagonists/changeling/powers/absorb.dm]: No

File: code/modules/antagonists/changeling/powers/absorb.dm (hunk 5 above)
Comment (PKPenguin321) [reply code/modules/antagonists/changeling/powers/absorb.dm]: Just curious why this method is better. Will implement it, but I had assumed that asking for a specific path in the for() would only return that path, so functionality shouldn't change?

**Pull request merged**